### PR TITLE
feat(cli-plugin-scaffold-react-component): scaffold to create react comp

### PR DESCRIPTION
**Pull request by brunozoric**
## Related Issue
We need a scaffold that will create and prepare package for a simple react component.

## Your solution
Created the scaffold.

## How Has This Been Tested?
Locally in dev repo + verdaccio publish and install

### Scaffolding process
process

### index.tsx
index

### ButtonComponent.tsx
buttoncomponent

### CODE REVIEW

**Comment by adrians5j** [comment]
A screenshot might be useful (or even a GIF hehe) 🙂

File: packages/cli-plugin-scaffold-react-component/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/CHANGELOG.md]: Remove this file. All of the content below is irrelevant to the newly added package.

File: packages/cli-plugin-scaffold-react-component/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# @webiny/cli-plugin-scaffold-admin-app-module
+
+[![](https://img.shields.io/npm/dw/@webiny/cli-plugin-scaffold-admin-app-module.svg)](https://www.npmjs.com/package/@webiny/cli-plugin-scaffold-admin-app-module)
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/README.md]: Check contents of this file.

File: packages/cli-plugin-scaffold-react-component/package.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    "pluralize": "^8.0.0",
+    "replace-in-path": "^1.0.4",
+    "validate-npm-package-name": "^3.0.0",
+    "write-json-file": "^4.3.0"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.5.5",
+    "@babel/core": "^7.5.5",
+    "@babel/preset-env": "^7.5.5",
+    "@babel/preset-flow": "^7.0.0",
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/package.json]: `preset-flow` ? 

Check the contents of this file.

File: packages/cli-plugin-scaffold-react-component/template/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  "dependencies": {
+    "react": "^16.14.0"
+  },
+  "devDependencies": {
+    "@babel/cli": "^7.5.5",
+    "@babel/core": "^7.5.5",
+    "@types/react": "^16.4.0",
+    "rimraf": "^3.0.2",
+    "typescript": "^4.1.3",
+    "jest-dynalite": "^3.3.1",
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/template/package.json]: jest-dynalite? Check the contents of this file.

**Review by adrians5j — CHANGES_REQUESTED** [review]
Please check the comments I left. Minor stuff.

File: packages/cli-plugin-scaffold-react-component/template/src/Target.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from "react";
+
+const colors = ["#FF0000", "#F0F0F0", "#FF00FF", "#FFFFFF", "#00FFFF", "#0000FF"];
+
+interface Style {
+    backgroundColor?: string;
+}
+
+interface Props {
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/template/src/Target.tsx:3]: While I admire your will to make a cool example, I think we can go with even something more simple, for example:

```tsx
import React, { useState } from "react";

interface Props {
  id?: string;
}

const Target: React.FunctionComponent<Props> = ({ id }) => {
  const [randomNumber, setRandomNumber] = useState<number>();

  return (
    <div>
      <div>ID: {id || "N/A"}</div>
      <div>Random Number: {randomNumber || "N/A"}</div>
      <div>
        <button onClick={() => setRandomNumber(Math.random())}>
          Generate Random Number
        </button>
      </div>
    </div>
  );
};

export default Target;
```

File: packages/cli-plugin-scaffold-react-component/.babelrc.js [hunk 6]
```diff
@@ -0,0 +1 @@
+module.exports = require("../../.babel.node")({ path: __dirname });
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/.babelrc.js:1]: This needs to be `babel.react`, not `babel.node`.

File: packages/cli-plugin-scaffold-react-component/package.json [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  "version": "5.2.1",
+  "description": "Generate a simple React component package.",
+  "main": "index.js",
+  "types": "index.d.ts",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/webiny/webiny-js.git",
+    "directory": "packages/cli-plugin-scaffold-react-component"
+  },
+  "author": "",
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/package.json]: Please use:
```
 "author": {
    "name": "Webiny",
    "url": "https://www.webiny.com"
  },
  ```

File: packages/cli-plugin-scaffold-react-component/template/package.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  "license": "MIT",
+  "scripts": {
+    "build": "rimraf ./dist '*.tsbuildinfo' && babel src -d dist --source-maps --copy-files --extensions \".ts,.tsx\" && yarn postbuild",
+    "watch": "babel src -d dist --source-maps --copy-files --extensions \".ts,.tsx\" --watch",
+    "postbuild": "cp package.json dist/ && tsc -p tsconfig.build.json"
+  },
+  "dependencies": {
+    "react": "^16.14.0"
+  },
+  "devDependencies": {
```
Comment (adrians5j) [inline packages/cli-plugin-scaffold-react-component/template/package.json:14]: Check dev dependencies here. For example, I can see `"@babel/preset-react"` missing.

**Review by adrians5j — CHANGES_REQUESTED** [review]
Please check a couple of minor issues.

File: packages/cli-plugin-scaffold-react-component/.babelrc.js (hunk 6 above)
Comment (brunozoric) [reply packages/cli-plugin-scaffold-react-component/.babelrc.js:1]: This should be `babel.node` since it is for building the scaffold.

**Pull request merged**